src/stan/lang/generator/generate_array_builder_adds.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef STAN_LANG_GENERATOR_GENERATE_ARRAY_BUILDER_ADDS_HPP
+#define STAN_LANG_GENERATOR_GENERATE_ARRAY_BUILDER_ADDS_HPP
+
+#include <stan/lang/ast.hpp>
+#include <stan/lang/generator/generate_expression.hpp>
+#include <ostream>
+#include <vector>
+
+namespace stan {
+  namespace lang {
+
+    /**
```

Comment:
I'm with Sean that we need an `internal` namespace where we can tuck these things without trying to give them heavy doc or testing.  Otherwise, it just discourages us from pulling out functions because of all the boilerplate overhead.  I'd argue that more complex functions might need doc or testing, but this one's just a trivial pass of everything to avoid duplicating loops and code generation.